pkg/netpol/eval/internal/k8s/netpol.go

```diff
@@ -512,79 +533,61 @@ func (np *NetworkPolicy) scanEgressRules() ([]SingleRuleLabels, error) {
-		err = np.updateNetworkPolicyGeneralConn(true, true, rulePorts, isIngress)
+		err = np.updateNetworkPolicyWideExposureConns(true, true, rulePorts, isIngress)
 		return nil, err
 	}
 	for i := range rules {
-		var ruleSel SingleRuleLabels
+		var ruleSel SingleRuleSelectors
 		if rules[i].IPBlock != nil {
 			continue
 		}
-		// note: LabelSelectorAsMap returns a nil map for nil selector,
-		// and returns empty map (map[string]string{}) for empty selector (all)
-		nsSelectorMap, err := metav1.LabelSelectorAsMap(rules[i].NamespaceSelector)
-		if err != nil {
-			return nil, err
-		}
-		podSelectorMap, err := metav1.LabelSelectorAsMap(rules[i].PodSelector)
-		if err != nil {
-			return nil, err
-		}
 		// a rule is exposed to entire cluster if :
-		// 1. the podSelector is nil (no podselector) but the namespaceSelector is empty ({})
+		// 1. the podSelector is nil (no podSelector) but the namespaceSelector is empty ({}) not nil
 		// 2. both podSelector and namespaceSelector are empty ({})
-		if (rules[i].PodSelector == nil && isEmptyMap(nsSelectorMap)) ||
-			(isEmptyMap(nsSelectorMap) && isEmptyMap(podSelectorMap)) {
-			err = np.updateNetworkPolicyGeneralConn(false, true, rulePorts, isIngress)
+		// (note that podSelector and namespaceSelector cannot be both nil, this is invalid )
+		// if podSelector is not nil but namespaceSelector is nil, this is the netpol's namespace
+		if rules[i].NamespaceSelector != nil && rules[i].NamespaceSelector.Size() == 0 &&
+			(rules[i].PodSelector == nil || rules[i].PodSelector.Size() == 0) {
+			err = np.updateNetworkPolicyWideExposureConns(false, true, rulePorts, isIngress)
 			return nil, err
 		}
-		// else (selector not empty)
-		ruleSel.PodLabels = podSelectorMap
-		// special case: ns selector is nil but podSelector is not, so the namespace of the rule is
-		// the policy's namespace; adding the k8s namespace name key to ruleSel.NsLabels
-		if rules[i].NamespaceSelector == nil && rules[i].PodSelector != nil {
-			ruleSel.PolicyNsFlag = true
-			ruleSel.NsLabels = map[string]string{common.K8sNsNameLabelKey: np.Namespace}
-		} else {
-			ruleSel.NsLabels = nsSelectorMap
-		}
+		// else selectors' combination specifies workloads by labels (at least one is not nil and not empty)
+		ruleSel.PodSelector = rules[i].PodSelector
+		ruleSel.NsSelector = rules[i].NamespaceSelector
 		rulesSelectors = append(rulesSelectors, ruleSel)
 	}
 	return rulesSelectors, nil
 }
 
-// isEmptyMap gets a labels map and returns if it is not nil but empty map
-func isEmptyMap(labelsMap map[string]string) bool {
-	return labelsMap != nil && len(labelsMap) == 0
-}
-
-func (np *NetworkPolicy) updateNetworkPolicyGeneralConn(allDests, entireCluster bool, rulePorts []netv1.NetworkPolicyPort,
+// updateNetworkPolicyWideExposureConns updates the wide exposure connections of the policy
+func (np *NetworkPolicy) updateNetworkPolicyWideExposureConns(externalExposure, entireCluster bool, rulePorts []netv1.NetworkPolicyPort,
```

Comment:
rename func to `updateNetworkPolicyExposureClusterWideConns`

Reply:
done